ISOLATION.md

```diff
@@ -0,0 +1,49 @@
+## SDN with Isolation for Openshift
+
+This document describes the architecture, benefits, and limitations of the isolation/multitenant implementation for openshift-sdn.
+
+#### Network Architecture
+
+The main components on a node are:
+
+* **br0** - the Open vSwitch bridge, which handles all isolation tasks through OpenFlow rules
+* **vethXXXXX** - the veth pairs that connect a pod's network namespace to the OVS bridge
+* **tun0** - an OVS internal port assigned the OpenShift node gateway address (10.1.x.1/24) for outside network communication.  iptables rules NAT traffic from tun0 to the outside network.
+* **lbr0** - the docker bridge, handles IPAM for all docker containers and OpenShift pods
+* **vovsbr**/**vlinuxbr** - veth pair that connects the docker bridge (lbr0) to the OVS bridge, to allow docker-only containers to talk to OpenShift pods and to access the outside network through tun0
+* **vxlan0** - an OVS VXLAN tunnel for communication with all other cluster nodes; directed to destination node with OF rules
+
+See `isolation-node-interfaces-diagram.pdf` for a diagram of how all these interfaces relate to each other.
+
+#### IP Address allocation
+
+When using Vagrant to configure the cluster, the master and each node are assigned cluster-private IP addresses in the 10.245.2.x/24 subnet through statements in the Vagrantfile.  The master receives 10.245.2.2 and the minions recieve an address above that.  The cluster uses these 10.245.2.x addresses for all communication between master and nodes.
+
+Regardless of how the cluster-public IP addresses are allocated (either via Vagrant, manual configuration, or some other provisioning system), the IP addresses of the master and each node are hardcoded in the master's openshift-master systemd startup script, configuration files passed to each node's openshift node process with --config, each node's 'kubeconfig' (referenced from the nodes --config YAML file), and each node's /etc/hosts file.
```

Comment:
The IP addresses of the nodes are not hardcoded in the master's configuration file. It may be done in the vagrant setup to jumpstart a cluster, but certainly not required.